core/dbt/contracts/graph/unparsed.py

```diff
@@ -484,8 +485,21 @@ class UnparsedMetric(dbtClassMixin, Replaceable):
     @classmethod
     def validate(cls, data):
         super(UnparsedMetric, cls).validate(data)
-        if "name" in data and " " in data["name"]:
-            raise ParsingException(f"Metrics name '{data['name']}' cannot contain spaces")
+        if "name" in data:
+            errors = []
+            if " " in data["name"]:
+                errors.append("cannot contain spaces")
+            if len(data["name"]) > 126:
+                errors.append("cannot contain more than 126 characters")
+            if not (re.match(r"^[A-Za-z]", data["name"])):
+                errors.append("must begin with a letter")
+            if not (re.match(r"[\w-]+$", data["name"])):
+                errors.append("must contain only letters, numbers and underscores")
+
+            if errors:
+                raise ParsingException(
+                    f"Metrics name '{data['name']}' is invalid.  It {', '.join(e for e in errors)}"
+                )
```

Comment:
@callum-mcdata would love some help with error messaging here!  I went with concatenating all the errors together but open to other solutions.

Reply:
Love the framework here! I'd only tweak it slightly to `The metric name` as opposed to `Metrics name`. Apart from that I think it is does a great job of describing the error and I LOVE that it provides all the errors in one failed compile instead of one at a time like I might have novicely implemented it